src/js/components/DataFilter/DataFilter.js

```diff
@@ -108,15 +127,21 @@ export const DataFilter = ({
         content = (
           <CheckBoxGroup id={id} name={property} options={booleanOptions} />
         );
-      } else if (options.length < 7) {
+      } else if (options.length < maxCheckBoxGroupOptions) {
         content = <CheckBoxGroup id={id} name={property} options={options} />;
       } else {
         content = (
           <SelectMultiple
             id={id}
             name={property}
             showSelectedInline
-            options={options}
+            options={searchedOptions}
+            onSearch={
+              options.length > minSelectSearchOptions
```

Comment:
If defined as "min" check should include the min, correct?

```suggestion
              options.length >= minSelectSearchOptions
```